src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -191,37 +190,37 @@ impl<'cg, 'cx, 'gcx, 'tcx> Visitor<'tcx> for ConstraintGeneration<'cg, 'cx, 'gcx
 impl<'cx, 'cg, 'gcx, 'tcx> ConstraintGeneration<'cx, 'cg, 'gcx, 'tcx> {
     /// The MIR type checker generates region liveness constraints
     /// that we also have to respect.
-    fn add_region_liveness_constraints_from_type_check(
-        &mut self,
-        liveness_set: &[(ty::Region<'tcx>, Location)],
-    ) {
-        debug!(
-            "add_region_liveness_constraints_from_type_check(liveness_set={} items)",
-            liveness_set.len(),
-        );
-
+    fn add_region_liveness_constraints_from_type_check(&mut self) {
         let ConstraintGeneration {
             regioncx,
             location_table,
             all_facts,
             ..
         } = self;
 
-        for (region, location) in liveness_set {
-            debug!("generate: {:#?} is live at {:#?}", region, location);
-            let region_vid = regioncx.to_region_vid(region);
-            regioncx.add_live_element(region_vid, *location);
-        }
+        debug!(
+            "add_region_liveness_constraints_from_type_check(liveness_constraints={} items)",
+            regioncx.number_of_liveness_constraints(),
+        );
 
         if let Some(all_facts) = all_facts {
-            all_facts
-                .region_live_at
-                .extend(liveness_set.into_iter().flat_map(|(region, location)| {
-                    let r = regioncx.to_region_vid(region);
-                    let p1 = location_table.start_index(*location);
-                    let p2 = location_table.mid_index(*location);
-                    iter::once((r, p1)).chain(iter::once((r, p2)))
-                }));
+            for (r, _) in regioncx.liveness_constraints() {
```

Comment:
We could also generate these during typeck -- we generate other facts then iirc

Reply:
Fixed this.